automation/jenkins/aws/setCredentials.sh

```diff
@@ -6,78 +6,152 @@
-                --role-arn arn:aws:iam::${!AWS_CRED_AWS_ACCOUNT_ID_VAR}:role/${AWS_CRED_AUTOMATION_ROLE} \
-                --role-session-name "$(echo $GIT_USER | tr -cd '[[:alnum:]]' )" \
-                --duration-seconds "${AUTOMATION_ROLE_VALIDITY}" \
-                --output json > ${TEMP_CREDENTIAL_FILE} ||
-                aws sts assume-role \
-                    --role-arn arn:aws:iam::${!AWS_CRED_AWS_ACCOUNT_ID_VAR}:role/${AWS_CRED_AUTOMATION_ROLE} \
-                    --role-session-name "$(echo $GIT_USER | tr -cd '[[:alnum:]]' )" \
-                    --output json > ${TEMP_CREDENTIAL_FILE}
-            unset AWS_ACCESS_KEY_ID
-            unset AWS_SECRET_ACCESS_KEY
-            AWS_CRED_TEMP_AWS_ACCESS_KEY_ID=$(jq -r '.Credentials.AccessKeyId' < ${TEMP_CREDENTIAL_FILE})
-            AWS_CRED_TEMP_AWS_SECRET_ACCESS_KEY=$(jq -r '.Credentials.SecretAccessKey' < ${TEMP_CREDENTIAL_FILE})
-            AWS_CRED_TEMP_AWS_SESSION_TOKEN=$(jq -r '.Credentials.SessionToken' < ${TEMP_CREDENTIAL_FILE})
-            rm ${TEMP_CREDENTIAL_FILE}
+    azure)
+
+        # find the method - default to prompt based login
+        AZ_AUTOMATION_AUTH_METHOD_VAR="${CRED_ACCOUNT}_AZ_AUTH_METHOD"
+        if [[ -z "${!AZ_AUTOMATION_AUTH_METHOD_VAR}" ]]; then AZ_AUTOMATION_AUTH_METHOD_VAR="AZ_AUTH_METHOD"; fi
+        AZ_AUTH_METHOD="${!AZ_AUTOMATION_AUTH_METHOD_VAR:-interactive}"
+
+        # lookup the AZ Account
+        AZ_ACCOUNT_ID_VAR="${CRED_ACCOUNT}_AZ_ACCOUNT_ID"
+        if [[ -n "${!AZ_ACCOUNT_ID_VAR}" ]]; then
+            AZ_ACCOUNT_ID="${AZ_ACCOUNT_ID_VAR}"
         fi
-    fi
-fi
+
+        # Set the tenant - this can be account specific but most likely global
+        AZ_ACCOUNT_TENANT_OVERRIDE_VAR="${CRED_ACCOUNT}_AZ_TENANT_ID"
+        if [[ -n "${!AZ_ACCOUNT_TENANT_OVERRIDE_VAR}" ]]; then
+            AZ_TENANT_ID="${AZ_ACCOUNT_TENANT_OVERRIDE_VAR}"
+        fi
+
+        case "${AZ_AUTH_METHOD}" in
+            service)
+
+                #Account specific creds
+                AZ_CRED_OVERRIDE_USERNAME_VAR="${CRED_ACCOUNT}_AZ_USERNAME"
+                AZ_CRED_OVERRIDE_PASS_VAR="${CRED_ACCOUNT}_AZ_PASS"
+
+                if [[ ( -n "${!AZ_CRED_OVERRIDE_USERNAME_VAR}") && (-n "${!AZ_CRED_OVERRIDE_PASS_VAR}") ]]; then
+                    AZ_CRED_USERNAME="${AZ_CRED_OVERRIDE_USERNAME_VAR}"
+                    AZ_CRED_PASS="${AZ_CRED_OVERRIDE_PASS_VAR}"
+                else
+
+                    # Tenant wide credentials
+                    AZ_CRED_AUTOMATION_USERNAME_VAR="${CRED_ACCOUNT}_AUTOMATION_USER"
+                    if [[ -z "${!AZ_CRED_AUTOMATION_USERNAME_VAR}" ]]; then AZ_CRED_AUTOMATION_USERNAME_VAR="AZ_AUTOMATION_USER" ; fi
+
+                    if  [[ -n "${!AWS_CRED_AUTOMATION_USER_VAR}" ]]; then
+                        AZ_CRED_USERNAME="${AZ_CRED_AUTOMATION_USERNAME_VAR}_AZ_USERNAME"
+                        AZ_CRED_PASS="${AZ_CRED_AUTOMATION_USERNAME_VAR}_AZ_PASS"
+                    fi
```

Comment:
I'm not too clear on what this block does. Is `AWS` intended? I can't see how this value could be set. If correct, perhaps a comment would be appropriate?

Reply:
Copy and paste error